docs/message-routing.md

```diff
@@ -3,23 +3,28 @@ Message routing can be split up into the routing of incoming and outgoing `Messa
 
 ## Incoming `Messages`
```

Comment:
Add explanation on:

- interaction types: agent to agent and component to component and its implications on the message to/sender fields